opencti-platform/opencti-front/src/private/components/settings/settings_messages/SettingsMessagesBanner.tsx

```diff
@@ -0,0 +1,220 @@
+    if (newBannerHeight !== bannerHeight) {
+      setBannerHeight(computeBannerHeight());
+    }
+  });
+  return bannerHeight;
+};
+
+// -- FUNCTION COMPONENT --
+
+interface MessageFromLocalStorage {
+  id: string
+  message: string
+  activated: boolean
+  dismissible: boolean
+  updated_at: Date
+  dismiss: boolean
+}
+
+const SettingsMessagesBannerComponent = ({
+  queryRef,
+}: { queryRef: PreloadedQuery<SettingsMessagesBannerQuery> }) => {
+  const classes = useStyles();
+  const theme = useTheme<Theme>();
+  const { t } = useFormatter();
+
+  const { settings } = usePreloadedQuery<SettingsMessagesBannerQuery>(settingsMessagesQuery, queryRef);
+  const config = useMemo<GraphQLSubscriptionConfig<SettingsMessagesBannerSubscription>>(() => ({
+    subscription: settingsSubscription,
+    variables: { id: settings.id },
+  }), [settings, settingsSubscription]);
+  useSubscription(config);
+
+  const messagesSettings = settings.messages ?? [];
+  const [messagesLocalStorage, setMessagesLocalStorage] = useState<MessageFromLocalStorage[]>(getMessagesFromLocalStorage());
+
+  useEffect((() => {
+    dispatchEvent();
+  }), [messagesLocalStorage]);
+
+  // 1. No message
+  if ((messagesSettings.length ?? 0) === 0) {
+    // Reset local storage
+    if (messagesLocalStorage.length > 0) {
+      setMessagesToLocalStorage([]);
+      setMessagesLocalStorage(getMessagesFromLocalStorage());
+    }
+    return (<></>);
+  }
+
+  // 2. New message -> Update local storage
+  const messagesUpdated = messagesSettings.map((message) => {
+    const messageLocalStorage = messagesLocalStorage.find((m) => m.id === message.id);
+    if (messageLocalStorage) {
+      if (messageLocalStorage.updated_at < message.updated_at || messageLocalStorage?.dismissible !== message.dismissible) {
+        return { ...message, dismiss: false };
+      }
+      return messageLocalStorage;
+    }
+    return { ...message, dismiss: false };
+  });
+  if (!R.equals(messagesUpdated, messagesLocalStorage)) {
+    setMessagesToLocalStorage(messagesUpdated);
+    setMessagesLocalStorage(getMessagesFromLocalStorage());
+  }
+  // 3. Retrieve message to display
+  const [messageToDisplay] = extractMessagesToDisplay(messagesLocalStorage);
+
+  // 4. Message not activated or dismiss
+  if (!messageToDisplay) {
+    return (<></>);
+  }
+
+  // 5. Message activated and not dismiss
+  const handleDismiss = () => {
+    const idx = messagesLocalStorage.findIndex((m) => m.id === messageToDisplay.id);
+    messagesLocalStorage.splice(idx, 1);
+    setMessagesToLocalStorage([...messagesLocalStorage, { ...messageToDisplay, dismiss: true }]);
+    setMessagesLocalStorage(getMessagesFromLocalStorage());
+  };
+
+  return (
+    <div id={BANNER_DIV_KEY} className={classes.container}>
+      <div className={classes.message}>{messageToDisplay.message}</div>
+      {messageToDisplay.dismissible
+        && <Link component="button" onClick={handleDismiss} className={classes.button}>
```

Comment:
Suggestion for a style that could look better : 
```
button: {
    color: '#663c00',
    position: 'absolute',
    top: '5px',
    right: '8px',
  }
```
```suggestion
        && (
              <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  className={classes.button}
                  onClick={handleDismiss}
              >
                <Close fontSize="inherit" />
              </IconButton>
          )
```

Reply:
Great, that's much better. Thank you so much !